### PR TITLE
Minor updates for new S3 permission defaults

### PR DESCRIPTION
**Pull request by JimTharioAmazon**
*Description of changes:*
1. Explicitly set ObjectOwnership property for all logging buckets
2. Pin runtime version for API-G authorizer to ensure match with layer
4. Update support files and scripts for new build pipeline

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/core-api-authorizers-sample/chalice/merge_template.json [hunk 1]
```diff
@@ -612,6 +612,13 @@
             "UpdateReplacePolicy": "Retain",
             "Properties": {
                 "AccessControl": "LogDeliveryWrite",
+                "OwnershipControls": {
```
Comment (thyagram-aws) [inline source/core-api-authorizers-sample/chalice/merge_template.json:615]: We should NOT use objectOwnership: ObjectOwnership.OBJECT_WRITER as it overrides default for S3 to S3 use case
(according to quip doc)
Also, another recommendation is to remove the ACL from your code and add the policy instead.

**Review by thyagram-aws — APPROVED** [review]
thank you for the changes

File: source/core-api-authorizers-sample/chalice/merge_template.json (hunk 1 above)
Comment (JimTharioAmazon) [reply source/core-api-authorizers-sample/chalice/merge_template.json:615]: This change needs to accommodate S3, CloudFront and VPC access logging. The latest CloudFront documentation states that ACLs must be enabled for its logging after the S3 defaults change. There is no bucket policy alternative for that. 

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html#AccessLogsBucketAndFileOwnership

S3 can use either ACLs or a bucket policy for logging, and VPC logs always use a bucket policy.

**Review by tomgilman — APPROVED** [review]
Looks good to me.

**Pull request merged**